packages/block-editor/src/components/block-list/use-block-props/use-focus-first-element.js

```diff
@@ -95,6 +95,24 @@ export function useFocusFirstElement( clientId ) {
 			return;
 		}
 
+		// Check to see if Block contains focussable element before a generic caret insert.
+		if ( ! ref.current.getAttribute( 'contenteditable' ) ) {
+			const whitelistTags = [ 'a', 'button' ];
+			let focusElements = focus.tabbable.findNext( ref.current );
+			// In the event a Block has children, focus will find the next element with tabindex which would happen to be the wrapping element of a child Block. This ensures we only focus on whitelisted tag such as link/button.
```

Comment:
What happened here was this.

1. Open the Block Editor.
2. Insert an Image Block.
3. Convert to Group Block from the Image Block Toolbar.
4. When focus was placed on the Group Block, the code found the next focussable element which happens to be the Image Block wrapper.
5. I adjusted the code so it must look for a whitelisted tag. This way focus will skip the Image Block wrapper and land on the first button/link contained within the Image Block.

I tried to sort this out with .filter() but I could just never get the node sorting to work properly.